src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,434 @@
+      // key: criteriaGroup.name,
+      style: {
+        backgroundColor: 'white',
+        borderRadius: '5px',
+        marginTop: 5,
+        border: `1px solid ${colors.dark(0.35)}`,
+      },
+    },
+    [
+      div({ style: { padding: '1rem' } }, [
+        div(
+          {
+            style: {
+              display: 'flex',
+              width: '100%',
+              justifyContent: 'space-between',
+              alignItems: 'baseline',
+            },
+          },
+          [
+            div(
+              {
+                style: {
+                  display: 'flex',
+                  justifyContent: 'space-between',
+                  alignItems: 'baseline',
+                },
+              },
+              [
+                h(Select, {
+                  'aria-label': 'must or must not meet',
+                  options: ['Must', 'Must not'],
+                  value: criteriaGroup.mustMeet ? 'Must' : 'Must not',
+                  onChange: () => updateCohort(_.set(`criteriaGroups.${index}.mustMeet`, !criteriaGroup.mustMeet)),
+                }),
+                div({ style: { margin: '0 10px' } }, ['meet']),
+                h(Select, {
+                  'aria-label': 'all or any',
+                  styles: { container: (provided) => ({ ...provided, style: { marginLeft: 10 } }) },
+                  options: ['any', 'all'],
+                  value: criteriaGroup.meetAll ? 'all' : 'any',
+                  onChange: () => updateCohort(_.set(`criteriaGroups.${index}.meetAll`, !criteriaGroup.meetAll)),
+                }),
+                div({ style: { marginLeft: 10 } }, ['of the following criteria:']),
+              ]
+            ),
+            div({ style: { alignItems: 'center', display: 'flex' } }, [
+              strong({ style: { marginRight: 5, fontSize: 16 } }, [criteriaGroup.name]),
+              h(
+                Link,
+                {
+                  'aria-label': 'delete group',
+                  onClick: () =>
+                    updateCohort(_.set('criteriaGroups', _.without([criteriaGroup], cohort.criteriaGroups))),
+                },
+                [icon('trash')]
+              ),
+            ]),
+          ]
+        ),
+        div([
+          (criteriaGroup.criteria.length !== 0 &&
+            _.map(
+              renderCriteriaView((criteria: AnyCriteria) =>
+                updateCohort(_.set(`criteriaGroups.${index}.criteria`, _.without([criteria], criteriaGroup.criteria)))
+              ),
+              criteriaGroup.criteria
+            )) ||
+            div({ style: { marginTop: 15 } }, [
+              div({ style: { fontWeight: 'bold', fontStyle: 'italic' } }, ['No criteria yet']),
+              div({ style: { fontStyle: 'italic', marginTop: 5 } }, [
+                "You can add a criteria by clicking on 'Add criteria'",
+              ]),
+            ]),
+        ]),
+        div({ style: { marginTop: 10 } }, [
+          h(GroupedSelect<CriteriaType>, {
```

Comment:
nice job in general of breaking up larger components into a composition of smaller sub-components.... Breaking out this GroupedSelect into its own more specialized CohortCriteriaSelector could offload some of this bulk to that sub-component as well.